buildSrc/src/main/groovy/org/terasology/gradlegoo/GooPlugin.groovy

```diff
@@ -0,0 +1,68 @@
+package org.terasology.gradlegoo
+
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.artifacts.ComponentMetadataContext
+import org.gradle.api.artifacts.ComponentMetadataRule
+import org.gradle.api.artifacts.ResolutionStrategy
+import org.gradle.api.artifacts.component.ModuleComponentIdentifier
+
+
+@SuppressWarnings("unused")
+class GradleGooPlugin implements Plugin<Project> {
+    @Override
+    void apply(Project target) {
+        target.extensions.create("gradlegoo", GradleGooExtension)
+        target.dependencies {
+            components {
+                withModule("org.spf4j:spf4j-slf4j-test", Slf4jImplementationRule)
+            }
+        }
+    }
+}
+
+
+class Slf4jImplementationRule implements ComponentMetadataRule {
+    void execute(ComponentMetadataContext context) {
+        context.details.allVariants {
+            withCapabilities {
+                addCapability "logging", "slf4j-impl-capability", "0"
+            }
+        }
+    }
+}
+
+
+class GradleGooExtension {
+    Class<Slf4jImplementationRule> Slf4jImplementationRule = Slf4jImplementationRule
```

Comment:
stuff on this class will be available as attributes on this plugin when accessed by build.gradle.

Since we've automatically applied this rule on plugin activation to the only package we care about, we don't strictly need to export it, but could be handy to have exported in case we need to label something else this way.